ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -852,19 +854,9 @@ withConnectionManager ConnectionManagerArguments {
                 DuplexState {} ->
                   throwSTM (withCallStack (ImpossibleState peerAddr))
 
-                TerminatingState {} -> do
-                  let connState' = InboundIdleState
-                                     connId connThread handle
-                                     (connectionDataFlow version)
-                  writeTVar connVar connState'
-                  return (mkTransition connState connState')
+                TerminatingState {} -> return Nothing
```

Comment:
It's not clear to me if this is a problem with the testing environment or the connection manager itself.  

Hmm, I think I see now: when we are in `TerminatingState` or `TerminatedState` the socket might be already closed so we should not try to use it.  Would you agree with that?

This is a nice example bug caught by the testing framework, please create a jira ticket for it and link it back here.

Reply:
Exactly so testing found two problems to be related to weird race conditions and the cause was a wrong assumption that the `connVar` could be reused if in the `Terminating` or `TerminatedState`.

I am not creating jira tickets but GitHub issues and linking them to Jira, is that alright?

Reply:
> I am not creating jira tickets but GitHub issues and linking them to Jira, is that alright?

That's alright, maybe add some label so we can easily find them, e.g. `discovered-by-quickcheck` or something like that.